### PR TITLE
[14.0][ADD] l10n_th_account_statement_import_kbiz: Statement import for Kasikorn KBiz

### PR DESCRIPTION
**Pull request by rossigee**
Looking for comments and feedback on this module I developed to import the CSV statement files I download from Kasikorn KBiz.

### CODE REVIEW

File: l10n_th_account_statement_import_kbiz/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+    'name': 'Import Kasikorn KBiz Bank Statement',
+    'category': 'Banking addons',
+    'version': '14.0.1.1.0',
+    'license': 'AGPL-3',
+    'author': 'Ross Golder',
```
Comment (Saran440) [inline l10n_th_account_statement_import_kbiz/__manifest__.py]:
```suggestion
    'author': 'Ross Golder, Odoo Community Association (OCA)',
```

File: l10n_th_account_statement_import_kbiz/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+    'name': 'Import Kasikorn KBiz Bank Statement',
+    'category': 'Banking addons',
+    'version': '14.0.1.1.0',
```
Comment (Saran440) [inline l10n_th_account_statement_import_kbiz/__manifest__.py]:
```suggestion
    'version': '14.0.1.0.0',
```

File: l10n_th_account_statement_import_kbiz/models/parser.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+import logging
+import hashlib
+import datetime
+
+from odoo import _, models
+
+_logger = logging.getLogger(__name__)
+
+
+def _prepare_transaction_line_kbiz_type1(invals):
```
Comment (Saran440) [inline l10n_th_account_statement_import_kbiz/models/parser.py]: All function should move in `account.statement.import.kbiz.parser` for easy inherit.

File: l10n_th_account_statement_import_kbiz/tests/test_files/type1-th.csv [hunk 4]
```diff
@@ -0,0 +1,21 @@
+﻿รายการเดินบัญชีเงินฝากออมทรัพย์ (มีรายละเอียด),,,,,,,,,,,,
```
Comment (Saran440) [inline l10n_th_account_statement_import_kbiz/tests/test_files/type1-th.csv:1]: move `test_files` to `demo` folder

- l10n_th_account_statement_import_kbiz
   |_ demo
   |_ models
   |_ tests

**Review by Saran440 — COMMENTED** [review]
Hi, thank you for your contribute in l10n-thailand

Can you use `pre-commit` for OCA pattern https://github.com/OCA/maintainer-tools/wiki/Install-pre-commit please.

I never import statement with KBiz. So, I have 2 question
1. Are there 2 format (csv, json)?
2. Where is official format bank?
3. I see `type1-th.csv` have 1/1 page. If statement has many page (1/3). What will happen and format is change?

Currenly, we use https://github.com/OCA/bank-statement-import/tree/14.0/account_statement_import_txt_xlsx and manual mapping for import statement
it easy to no code but not support json file. This module can help it.

File: l10n_th_account_statement_import_kbiz/tests/test_files/type1-th.csv (hunk 4 above)
Comment (rossigee) [reply l10n_th_account_statement_import_kbiz/tests/test_files/type1-th.csv:1]: The files were only intended for regression testing purposes. They probably aren't appropriate as demo data.

**Comment by rossigee** [comment]
> I never import statement with KBiz. So, I have 2 question
> 
>     1. Are there 2 format (csv, json)?
>

You can download CSV and PDF statements. There is no JSON option.

The JSON files included are for use by the regression tests to confirm the result.
 
>     2. Where is official format bank?
> 

I have no idea :shrug: 

>     3. I see `type1-th.csv` have 1/1 page. If statement has many page (1/3). What will happen and format is change?
> 

Good point. I don't have any statements that go into multiple pages that I can test with.

> Currenly, we use https://github.com/OCA/bank-statement-import/tree/14.0/account_statement_import_txt_xlsx and manual mapping for import statement it easy to no code but not support json file. This module can help it.

I'm not sure if that could be used here as the CSV file provided isn't the typical type with a simple-to-parse structure (i.e. header row then data rows). It's not really a proper CSV file.

**Pull request closed without merging**